### ISSUE TITLE
Gather example with axis=1 not correct

### ISSUE THREAD

**Issue by pulzar** [onnx/onnx#2346]
In the operators spec, the example for Gather op with axis=1 produces an incorrect shape:

[https://github.com/onnx/onnx/blob/master/docs/Operators.md#Gather](https://github.com/onnx/onnx/blob/master/docs/Operators.md#Gather)

```
  data = [
      [1.0, 1.2, 1.9],
      [2.3, 3.4, 3.9],
      [4.5, 5.7, 5.9],
  ]
  indices = [
      [0, 2],
  ]
  axis = 1,
  output = [
      [
          [1.0, 1.9],
          [2.3, 3.9],
          [4.5, 5.9],
      ],
  ]
```

The output shape in this example is (1, 3, 2). However, I believe the correct output is (3, 1, 2):

```
>>> data = [
...       [1.0, 1.2, 1.9],
...       [2.3, 3.4, 3.9],
...       [4.5, 5.7, 5.9],
...   ]
>>> indices = [
...       [0, 2],
...   ]
>>> np.take(data, indices, axis=1)
array([[[1. , 1.9]],

       [[2.3, 3.9]],

       [[4.5, 5.9]]])
>>> np.take(data, indices, axis=1).shape
(3, 1, 2)
```

**Comment by skottmckay** [comment]
Would it be better to simplify a little more as well and just have data with shape {3, 3}, indices with shape {2}, and output with shape {3, 2}?

  data = [
      [1.0, 1.2, 1.9],
      [2.3, 3.4, 3.9],
      [4.5, 5.7, 5.9]
  ]
  indices = [0, 2]
  axis = 1,
  output = [
      [1.0, 1.9],
      [2.3, 3.9],
      [4.5, 5.9]
  ]

**Comment by daquexian** [comment]
Fixed by #3040
